src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
I think it is possible to come up with a design that avoids shared pointer.

On my new fan project, where I am doing a refactor of the old fans.cc, I have been having good success with holding the object instances in a std::vector of unique_ptr that sits in the namespace just outside the class.  Then when you need to store how to access this fan elsewhere, just store the integer vector index.  The namespace has a little get index routine from the fan name but everything else is in the class.  Changing to zero-based vector indexing is not too bad.  

}; //class FanSystem 

extern std::vector < std::unique_ptr <FanSystem> > fanObjs;

} // Fan namespace

you can check out HVACFan.hh and HVACFan.cc here if you want
https://github.com/EnergyArchmage/EplusFanDev